content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md

```diff
@@ -0,0 +1,621 @@
+
+To integrate Arduino Cloud with Opta, it is essential to write Arduino Sketch code (based on C++) within the Opta PLC IDE. This code is crucial for establishing reliable communication, enabling the control and monitoring of Opta's variables, inputs, and outputs.
+
+The first step to ensure proper functionality is to add the necessary libraries. These libraries are collections of pre-written code that developers can integrate into their projects to perform specific functions or tasks. They offer ready-made solutions to common challenges, saving time and effort, and are essential for establishing effective communication between the Opta and the Arduino Cloud.
+
+### Required Libraries for Opta and Arduino Cloud Integration
+
+To ensure proper functionality and establish effective communication between the Opta and the Arduino Cloud, the following libraries and their respective minimum versions are required:
+
+| Library                        | Version | Description                                                                 |
+|--------------------------------|---------|-----------------------------------------------------------------------------|
+| **ArduinoIoTCloud**            | 1.15.1  | This library allows connecting to the Arduino IoT Cloud service. It provides a ConnectionManager to handle connection/disconnection, property-change updates and events callbacks. |
+| **Arduino_ConnectionHandler**   | 0.8.1   | Arduino Library for network connection management (WiFi, GSM, NB, [Ethernet]). |
+| **ArduinoECCX08**              | 1.3.8   | Arduino Library for the Atmel/Microchip ECC508 and ECC608 crypto chips. |
+| **ArduinoMqttClient**          | 0.1.8   | Allows you to send and receive MQTT messages using Arduino. |
+| **Arduino_DebugUtils**         | 1.4.0   | Debugging module with different debug levels, timestamps and printf-style output. This class provides functionality useful for debugging sketches via printf-style statements. |
+| **Arduino_Portenta_OTA**       | 1.2.1   | Firmware update for the Portenta H7. This library allows performing a firmware update on the Arduino Portenta H7. The firmware can be stored in various different locations such as within the microcontroller's flash, on an external SD card or on the QSPI flash chip. |
+| **Arduino_SecureElement**      | 0.1.2   | This library allows usage of Arduino boards secure elements in a common and unified way. |
+
+These libraries are indexed. They are certified to ensure optimal performance and reliability. Using them will accelerate your development process and strengthen the durability of your applications, especially in industrial environments. By incorporating these libraries with the specified versions, developers can efficiently establish and maintain secure and reliable communication between Opta and the Arduino Cloud, enabling robust control, monitoring, and remote updates.
+
+### Arduino Sketch Code
```

Comment:
Would be nice to have a screenshot of the PLC IDE so reader can mentally locate in the software where the code will be written.